.github/workflows/protocol-tests.yaml

```diff
@@ -10,37 +10,49 @@ jobs:
     name: Hardhat Tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
         with:
           node-version: '16.x'
+          cache: 'yarn'
+          cache-dependency-path: protocol/yarn.lock
+
       - run: yarn install
+        working-directory: protocol
+
       - run: yarn compile
         working-directory: protocol
+
       - run: yarn test
         working-directory: protocol
         env:
-          TESTS_MAINNET_RPC_URL: ${{ secrets.TESTS_MAINNET_RPC_URL }}
+          TESTS_MAINNET_RPC_URL: 'https://eth-mainnet.g.alchemy.com/v2/nXcOOw_AQOJ22NwPrpmjgTNuWMlK-Mds'
```

Comment:
@nichosystem Could you please add this secret into gh? I created a new Alchemy app, specifically for Mainnet CI tests:  https://dashboard.alchemy.com/apps/ybpk7yurqagxg1wp